src/main/java/com/puppycrawl/tools/checkstyle/checks/modifier/RedundantModifierCheck.java

```diff
@@ -225,30 +233,35 @@
             TokenTypes.CLASS_DEF,
             TokenTypes.ENUM_DEF,
             TokenTypes.RESOURCE,
+            TokenTypes.RECORD_DEF,
         };
     }
 
     @Override
     public void visitToken(DetailAST ast) {
-        if (ast.getType() == TokenTypes.INTERFACE_DEF) {
+        final int type = ast.getType();
+        if (type == TokenTypes.INTERFACE_DEF) {
             checkInterfaceModifiers(ast);
         }
-        else if (ast.getType() == TokenTypes.ENUM_DEF) {
+        else if (type == TokenTypes.ENUM_DEF) {
             checkEnumDef(ast);
         }
+        else if (type == TokenTypes.RECORD_DEF) {
+            checkRecordDef(ast);
+        }
         else {
-            if (ast.getType() == TokenTypes.CTOR_DEF) {
+            if (type == TokenTypes.CTOR_DEF) {
                 if (isEnumMember(ast)) {
                     checkEnumConstructorModifiers(ast);
                 }
                 else {
                     checkClassConstructorModifiers(ast);
                 }
             }
-            else if (ast.getType() == TokenTypes.METHOD_DEF) {
+            else if (type == TokenTypes.METHOD_DEF) {
                 processMethods(ast);
             }
-            else if (ast.getType() == TokenTypes.RESOURCE) {
+            else if (type == TokenTypes.RESOURCE) {
```

Comment:
Ok, refactor both `if-else` chain to switch? (main + `if-else` nested in `else`)

Reply:
yes. `switch` is designed exactly for cases like this - to avoid long if-else chains. Plus, here "nested" if else is nothing but a continuation of outer if-else chain.

Reply:
yes for refactoring both chains to switch **and** for extraction?

Reply:
yes, to both.

Reply:
it also may be a second commit with refactoring

Reply:
I changed it, kindly take a look.